nucypher/blockchain/eth/actors.py

```diff
@@ -1171,6 +1174,47 @@ def missing_commitments(self) -> int:
         missing = self.staking_agent.get_missing_commitments(checksum_address=staker_address)
         return missing
 
+    def _verify_policy_payment_by_policy_id(self, policy_id: str) -> bool:
+        from nucypher.policy.policies import Policy
+        arrangements = self.policy_agent.fetch_policy_arrangements(policy_id=policy_id[:Policy.POLICY_ID_LENGTH])
```

Comment:
Isn't policy id already pulled out by the BytestringSplitter in `server.py` in L246?

Reply:
This line kinda makes me wish policy ID was its own class